hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/StreamFileProducerConcurrent.java

```diff
@@ -182,64 +197,79 @@ public void switchBlocks(
      */
     @Override
     public void writeRecordStreamItems(@NonNull final Stream<SingleTransactionRecord> recordStreamItems) {
-        assert lastRecordHashingResult != null : "initRunningHash() must be called before writeRecordStreamItems";
-        assert currentRecordFileWriter != null : "switchBlocks() must be called before writeRecordStreamItems";
-        requireNonNull(recordStreamItems);
+        lock.lock(); // Block until the lock is acquired
 
-        // serialize all the record stream items in background thread into SerializedSingleTransaction objects
-        final var futureSerializedRecords = CompletableFuture.supplyAsync(
-                () -> recordStreamItems
-                        .map(item -> format.serialize(item, currentBlockNumber, hapiVersion))
-                        .toList(),
-                executorService);
-        // when serialization is done and previous running hash is computed, we can compute new running hash and write
-        // serialized items to record file in parallel update running hash in a background thread
-        lastRecordHashingResultNMinus3 = lastRecordHashingResultNMinus2;
-        lastRecordHashingResultNMinus2 = lastRecordHashingResultNMinus1;
-        lastRecordHashingResultNMinus1 = lastRecordHashingResult;
-        lastRecordHashingResult = lastRecordHashingResult
-                .thenCombine(futureSerializedRecords, TwoResults::new)
-                .thenApplyAsync(
-                        twoResults -> format.computeNewRunningHash(twoResults.a(), twoResults.b()), executorService);
-        // write serialized items to record file in a background thread
-        currentRecordFileWriter = currentRecordFileWriter
-                .thenCombine(futureSerializedRecords, TwoResults::new)
-                .thenApplyAsync(
-                        twoResults -> {
-                            final var writer = twoResults.a();
-                            final var serializedItems = twoResults.b();
-                            serializedItems.forEach(item -> {
-                                try {
-                                    writer.writeItem(item);
-                                } catch (final Exception e) {
-                                    logger.error("Error writing record item to file", e);
-                                }
-                            });
-                            return writer;
-                        },
-                        executorService);
+        try {
+            assert lastRecordHashingResult != null : "initRunningHash() must be called before writeRecordStreamItems";
```

Comment:
Both of these should probably be upgraded to full conditionals rather than test/debug-only assert.

Reply:
Good point. Probably good for a follow-up PR.